### PR TITLE
Fix/update draft

### PR DESCRIPTION
**Pull request by mithun30052001**
We need a save draft button on the edit draft page. Once the user opens the edit draft page, the only option available to the user is publish #128  error resolved

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -56,6 +54,8 @@ def update
       redirect_to community_path(@post.community_id)
     else
       @post.is_drafted = true
+      @post.update!(post_values)
```
Comment (iMagesh) [inline app/controllers/posts_controller.rb:57]: why do we have to update twice here? can you check and refactor?

File: app/views/communities/_form.html.erb [hunk 2]
```diff
@@ -4,15 +4,17 @@
     <div class="row">
       <div class="col-sm-12">
         <div class="create-post m-3">
-          <div class="form-group"> 
+          <div class="form-group">
             <%= label_tag :profile_picture %>
-            <%= form.file_field :profile_image, class: "form-control" %>
+            <%= form.file_field :profile_image, class: "form-control "+is_error(@community,(:profile_image)).to_s %>
```
Comment (iMagesh) [inline app/views/communities/_form.html.erb:9]: can you move the error part to a helper?

File: app/views/communities/_form.html.erb [hunk 3]
```diff
@@ -39,7 +41,8 @@
         <div class="create-post m-3">
           <div class="form-group">
             <%= label_tag :category %>
-            <%= form.select :category, Community::CATEGORIES, class: "form-control" %>
+            <%= form.select :category, Community::CATEGORIES, prompt: "Select Category" ,class: "form-control "+is_error(@community,(:category)).to_s %>
```
Comment (iMagesh) [inline app/views/communities/_form.html.erb:44]: you can move the error part to a helper?

File: app/views/posts/_form.html.erb [hunk 4]
```diff
@@ -15,29 +15,17 @@
...
-             <% if @post.errors[:body].any? %>
-              <div class="errors">
-                <%= form.text_field :title,  class: "form-control", placeholder: :title, maxlength: 20%>
-              </div>
-               <div class="error"><%= @post.errors[:title].first %></div>
-             <% else %>
-                <%= form.text_field :title, class: "form-control", placeholder: :title, maxlength: 20%>
-             <% end %>
+            <div class="form-group">
+              <%= form.text_field :title, class: "form-control "+is_error(@post,(:title)).to_s, placeholder: :title, maxlength: 20%>
```
Comment (iMagesh) [inline app/views/posts/_form.html.erb:19]: same for this

File: app/views/posts/_form.html.erb [hunk 5]
```diff
@@ -15,29 +15,17 @@
...
-              </div>
-              <div class="error"><%= @post.errors[:body].first %></div>
-              <% else %>
-              <%= form.rich_text_area :body, class: "textarea", placeholder: "Text(optional)" %>
-              <% end %>
-              <%#= form.rich_text_area :body, class: "textarea", placeholder: "Text(optional)" %>
-              <%= form.file_field :images, multiple: true, :class=>"form-control", :placeholder=>"" %>
+              <%= form.rich_text_area :body, class: "textarea "+is_error(@post,(:body)).to_s, placeholder: "Text(optional)" %>
+              <p class="text-danger"><%= error_msg(@post, :body)  %></p>
+              <%= form.file_field :images, multiple: true, :class=>"form-control "+is_error(@post,(:images)).to_s, :placeholder=>"" %>
```
Comment (iMagesh) [inline app/views/posts/_form.html.erb:27]: here as well

File: app/views/posts/_form.html.erb [hunk 6]
```diff
@@ -58,11 +46,13 @@
             </li>
           </ul>
           <div class="float-right">
-            <% unless params[:action] == 'edit' %>
             <div class="join-btn  create-post-btn mb-4">
-              <%= form.submit "Save as draft", class: "text-white" %>
+              <% if params[:action] == 'edit' %>
+                <%= form.submit "Update draft", class: "text-white" %>
+              <% else %>
+                <%= form.submit "Save as draft", class: "text-white" %>
```
Comment (iMagesh) [inline app/views/posts/_form.html.erb:53]: can you simplify this and avoid duplicates

**Review by iMagesh — CHANGES_REQUESTED** [review]
check comments

File: app/views/communities/_form.html.erb (hunk 3 above)
Comment (vasanthakumar-a) [reply app/views/communities/_form.html.erb:44]: same here also @iMagesh

File: app/views/communities/_form.html.erb (hunk 2 above)
Comment (iMagesh) [reply app/views/communities/_form.html.erb:9]: I was thinking we can move the "form-control" and .to_s part also into the helper method to make this look better. What do you think?

**Pull request merged**